### PR TITLE
Refactor to use the hugging face interface

### PR DESCRIPTION
**Pull request by AugustoPeres**
## Contents

This pull request refactors our code to work with the hugging face datasets interface. We completely remove the usage of the `torch.data.Dataset` class and use instead HuggingFace's interface.

## Technical debt

We accumulate technical debt in the normalization of the dataset. Previously dataset statistics where being computed inside the dataset class. Then, this statistics where given to the model where they were used to normalize the labels. When removing the `torch.Dataset` class from the code we **temporarily** lost this functionality. In fact, maybe I should have started by first re-implementing normalization. But I did not realize this until halfway through the refactoring. Additionally, pushing normalization further down allows us to fully test the hugging face interface right now.

The main evidence of this technical debt is where I hardcoded the statistics that are then fed to the model:

```python
    train_stats = {
        'x_mean': torch.tensor([0.0]),
        'x_std': torch.tensor([1.0]),
        'edge_attr_mean': torch.tensor([0.0]),
        'edge_attr_std': torch.tensor([1.0]),
        'y_mean': torch.tensor([0.0]),
        'y_std': torch.tensor([1.])
    }
```

I am fully aware of this issue and I have tagged the code whenever normalization and statistics gathering should happen. Implementing back this functionality is my **number one pripority**

### CODE REVIEW

File: meshnets/utils/model_training.py [hunk 1]
```diff
@@ -7,18 +7,14 @@
...
-from meshnets.modules.lightning_wrapper import MGNLightningWrapper
-from meshnets.modules.model import MeshGraphNet
-from meshnets.utils.callbacks import GeometricBatchSize
-from meshnets.utils.callbacks import GPUUsage
-from meshnets.utils.callbacks import GradientNorm
-from meshnets.utils.callbacks import MLFlowLoggerFinalizeCheckpointer
-
-import meshnets
-import datasets
+from .. import modules
```
[the remark this answers is not in the document]
Comment (AugustoPeres) [reply meshnets/utils/model_training.py:15]: I had it like this before. The problem is that `model_training.py` is part of the meshnets package:

```
meshnets
├── __init__.py
├── data_processing
│   ├── data_mappers.py
│   ├── __init__.py
│   ├── torch_utils.py
├── modules
│   ├── __init__.py
│   ├── lightning_wrapper.py
│   ├── model.py
└── utils
    ├── callbacks.py
    ├── __init__.py
    ├── model_training.py
```

Therefore importing meshnets was raising the `circular imports` warning. I don't think we should import `meshnets` inside `meshnets`. But please let me know if there is a better way to go about this

[the remark this answers is not in the document]
Comment (IvanPombo) [reply meshnets/utils/model_training.py:15]: Yes, but you are still techinically doing the same with your relative imports. In either case, notice that now you aren't importing meshnets with `import meshnets` or `import .` that might have been the root cause. Otherwise, you need to structure the inits to not have the circular imports, this also tells you that we may have problems with the structure of the code.

File: meshnets/utils/model_training.py [hunk 2]
```diff
@@ -75,15 +71,14 @@ def train_model(config):
                                           version=dataset_version,
                                           split='train')
     train_dataset = (train_dataset.map(
-        lambda x: meshnets.data_processing.data_mappers.to_undirected(
-            x, 'edges')).map(
-                meshnets.data_processing.data_mappers.make_edge_features).map(
-                    lambda x: meshnets.data_processing.data_mappers.
-                    make_node_features(x, 'wind_vector')))
+        lambda x: data_processing.data_mappers.to_undirected(x, 'edges')).map(
```
[the remark this answers is not in the document]
Comment (AugustoPeres) [reply meshnets/utils/model_training.py]: Certainly

File: meshnets/utils/model_training.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AugustoPeres) [reply meshnets/utils/model_training.py:15]: I will take a look into this and will report back

[the remark this answers is not in the document]
Comment (AugustoPeres) [reply meshnets/utils/model_training.py]: Removed the relative imports

**Review by IvanPombo — APPROVED** [review]
Thanks for fixing it. However, the callbacks is still relative. Gonna Approve it, but make sure to change that!
